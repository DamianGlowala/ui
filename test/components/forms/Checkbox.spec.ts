import { describe, it, expect } from 'vitest'
import { UCheckbox } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Checkbox', () => {
  it.each([
    ['basic case', {}],
    ['with custom id', { props: { id: 'custom-id' } }],
    ['with custom value', { props: { value: 'custom-value' } }],
    ['with custom name', { props: { name: 'custom-name' } }],
    ['disabled set to true', { props: { disabled: true } }],
    ['indeterminate set to true', { props: { indeterminate: true } }],
    ['with help text', { props: { help: 'Help text' } }],
    ['with label', { props: { label: 'Label' } }],
    ['required set to true', { props: { required: true } }],
    ['with custom color', { props: { color: 'blue' } }],
    ['with custom input class', { props: { inputClass: 'custom-input-class' } }],
    ['with custom class', { props: { class: 'custom-class' } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Checkbox.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UCheckbox)
    expect(html).toMatchSnapshot()
  })
})
