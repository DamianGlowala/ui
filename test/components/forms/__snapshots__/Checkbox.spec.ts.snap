// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > renders basic case correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders disabled set to true correctly 1`] = `
"<div class="">
  <div class=""><input disabled="" type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders indeterminate set to true correctly 1`] = `
"<div class="">
  <div class=""><input indeterminate="" type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders required set to true correctly 1`] = `
"<div class="">
  <div class=""><input required="" type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom class correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom color correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom id correctly 1`] = `
"<div class="">
  <div class=""><input id="custom-id" type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom input class correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class="custom-input-class"></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom modelValue correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom name correctly 1`] = `
"<div class="">
  <div class=""><input name="custom-name" type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with custom value correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class="" value="custom-value"></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with help text correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class=""></div>
  <!--v-if-->
</div>"
`;

exports[`Checkbox > renders with label correctly 1`] = `
"<div class="">
  <div class=""><input type="checkbox" class=""></div>
  <div class=""><label class="">Label
      <!--v-if-->
    </label>
    <!--v-if-->
  </div>
</div>"
`;
